File: 1281-subtract-the-product-and-sum-of-digits-of-an-integer/1281-subtract-the-product-and-sum-of-digits-of-an-integer.py
class Solution:
    def subtractProductAndSum(self, n: int) -> int:
        s = str(n)
        nums = [int(x) for x in s]
        _sum = sum(nums)
        _product = 1
        for num in nums:
            _product *= num

        return _product - _sum
        

File: 1281-subtract-the-product-and-sum-of-digits-of-an-integer/1281-subtract-the-product-and-sum-of-digits-of-an-integer.rs
impl Solution {
    pub fn subtract_product_and_sum(n: i32) -> i32 {
    let s= n.to_string();
    let  nums:Vec<i32>= s.chars().map(|x|x.to_string().parse().unwrap()).collect::<Vec<i32>>();

    let sum:i32= nums.clone().into_iter().sum();
    let mul:i32= nums.clone().into_iter().product();
        
     mul-sum

        
    }
}